targetselector/running_target_selector.go

```diff
@@ -153,6 +154,16 @@ func (r runningTargetSelector) TargetIsHealthy(target string, c client.Client, i
 		if p.Status.Phase != corev1.PodRunning {
 			return errors.New("pod is not Running")
 		}
+
+		// check if pod's node is gone in the case that this was a node failure
+		if instance.Spec.NodeFailure != nil {
+			var n corev1.Node
+			if err := c.Get(context.Background(), client.ObjectKey{Name: p.Spec.NodeName}, &n); err != nil {
+				if apierrors.IsNotFound(err) {
+					return err
+				}
```

Comment:
What does it mean to have an `err` that is not the `NotFound` error?

Reply:
It can be any kind of error while trying to get the resource (like apiserver timeout for instance).